### ISSUE TITLE
Checkerboard patterns in FullGrad

### ISSUE THREAD

**Issue by IgnacioHeredia** [idiap/fullgrad-saliency#16]
Hi all,

I had a question. When I use `Fullgrad` (or its smooth version) I experience very strong checkerboard patterns. I'm aware this is a fairly [well know issue](https://distill.pub/2016/deconv-checkerboard/) with gradient methods, but still curious because:
* `GradCam` implementation (from this very same repo) does not show them
* I don't see any mention of them in either the repo or the paper
*  I'm using a fairly standard architecture (ResNet18) from `torchvision.models`. No fancy stuff.

I'm skeptical of opening an issue because the patterns only appear in the model after training. The model with weights pretrained on ImageNet doesn't show these patterns. _Therefore it might very well be unrelated to Fullgrad_, so feel free to close the issue.

But I am still curious if someone else has had this problem, in case it could be a side effect of the [interpolation](https://github.com/idiap/fullgrad-saliency/blob/26fb91b2f9ef616d15d080644b9f8a656a68204f/saliency/fullgrad.py#L128) or whatnot.

### FullGrad
One can start to sense the patterns.

![fullgrad](https://user-images.githubusercontent.com/16413205/177504891-dc6fc50b-bea8-4788-8178-11eef40040c0.png)

### Smooth FullGrad
Averaging over the default 100 samples really magnifies the patterns.

![smooth_fullgrad](https://user-images.githubusercontent.com/16413205/177504977-8393e98c-626e-40f0-97ed-383e3329d6e7.png)

Of course, this behavior repeats for all images. Here's another example:

![smooth_fullgrad2](https://user-images.githubusercontent.com/16413205/177505091-6449522b-68e6-47bd-b85b-0f13714b583e.png)

### GradCam

No checkerboard, although one sees these reticular-like explanations. They also seem to appear in [other implementations](https://github.com/PAIR-code/saliency/blob/master/Examples_core.ipynb) thus are probably a distinctive feature of GradCam.

![gradcam](https://user-images.githubusercontent.com/16413205/177505339-1d088345-4d94-45e2-b362-04237d5c4ac9.png)

### Smooth FullGrad
No patterns.

![smfg](https://user-images.githubusercontent.com/16413205/177514845-a75f90b4-151f-4afe-aa06-8825a4dfd1a5.png)

### GradCam
No patterns, still kind of reticular.

![gradcam (1)](https://user-images.githubusercontent.com/16413205/177514942-edcc4fb5-a14f-4428-bfa8-1c6c03a5fbec.png)

So, just sharing in case anyone had any thoughts. Pointers are welcome, thanks! :)

**Comment by IgnacioHeredia** [comment]
#### Update

Training on a cleaner/smaller dataset with a `MobileNet` does not seem to produce patterns. So it probably  (?) had to do with the noisiness of the dataset. Still no intuition on why a noisy dataset would be more likely to produce these patterns.

Feel free to close the issue, though still interested to hear if someone has ever experienced something similar.
